AK/IntrusiveRedBlackTree.h

```diff
@@ -143,7 +143,7 @@ class IntrusiveRedBlackTree final : public BaseRedBlackTree<K> {
 
     static V* node_to_value(TreeNode& node)
     {
-        return (V*)((u8*)&node - ((u8*)&(((V*)nullptr)->*member) - (u8*)nullptr));
+        return (V*)((u8*)&node - (u8*)&(((V*)nullptr)->*member));
```

Comment:
Could we instead do the same as what `IntrusiveList` does, and avoid the C-style casts altogether?

https://github.com/SerenityOS/serenity/blob/master/AK/IntrusiveList.h#L400

Reply:
Sure. This change wasn't initially targeting that, but getting rid of C-style casts is something that I am particularly fond of.